cpp/perspective/CMakeLists.txt

```diff
@@ -586,16 +585,24 @@ message("${BUILD_MESSAGE}\n")
 if (PSP_WASM_BUILD)
 	add_library(psp ${WASM_SOURCE_FILES})
 	target_compile_definitions(psp PRIVATE PSP_ENABLE_WASM=1)
-	set_target_properties(psp PROPERTIES COMPILE_FLAGS "${ASYNC_MODE_FLAGS}")
+	set_target_properties(psp PROPERTIES COMPILE_FLAGS "")
 	target_link_libraries(psp arrow)
 
-	add_executable(perspective.async src/cpp/emscripten.cpp)
-	target_link_libraries(perspective.async psp "${ASYNC_MODE_FLAGS}")
-	target_compile_definitions(perspective.async PRIVATE PSP_ENABLE_WASM=1)
-
-	set_target_properties(perspective.async PROPERTIES COMPILE_FLAGS "${ASYNC_MODE_FLAGS}")
-	set_target_properties(perspective.async PROPERTIES RUNTIME_OUTPUT_DIRECTORY "./build/")
-	set_target_properties(perspective.async PROPERTIES OUTPUT_NAME "psp.async")
+	# "esm/erspective.cpp.js" from CMAKE_EXECUTABLE_SYNTAX
+	add_executable(perspective_esm src/cpp/emscripten.cpp)
+	target_link_libraries(perspective_esm psp "-s ENVIRONMENT=worker -s EXPORT_ES6=1")
+	target_compile_definitions(perspective_esm PRIVATE PSP_ENABLE_WASM=1)
+	set_target_properties(perspective_esm PROPERTIES COMPILE_FLAGS "-s ENVIRONMENT=worker -s EXPORT_ES6=1")
+	set_target_properties(perspective_esm PROPERTIES RUNTIME_OUTPUT_DIRECTORY "./esm/")
+	set_target_properties(perspective_esm PROPERTIES OUTPUT_NAME "perspective.cpp")
+
+	# "cjs/perspective.cpp.js" from CMAKE_EXECUTABLE_SYNTAX
+	add_executable(perspective_cjs src/cpp/emscripten.cpp)
+	target_link_libraries(perspective_cjs psp "-s ENVIRONMENT=node")
+	target_compile_definitions(perspective_cjs PRIVATE PSP_ENABLE_WASM=1)
+	set_target_properties(perspective_cjs PROPERTIES COMPILE_FLAGS "-s ENVIRONMENT=node")
+	set_target_properties(perspective_cjs PROPERTIES RUNTIME_OUTPUT_DIRECTORY "./cjs/")
+	set_target_properties(perspective_cjs PROPERTIES OUTPUT_NAME "perspective.cpp")
```

Comment:
Up until now, we've been building Emscripten's mixed JavaScript runtime, which supprots a variety of [states](https://github.com/emscripten-core/emscripten/blob/master/src/settings.js#L546) at the same time.  This is handy for a simple build, but leaves some uncalled references to technically invalid JS in the build which then must be configured in webpack.  Since we want to minimize the configuration required in webpack to the bare minimum, we build 2 Emscripten builds, with `"worker"` and `"node"` modes, as well as stripping ES modules from the node version since Node doesn't support them.